crypto/evp/ec_support.c

```diff
@@ -0,0 +1,205 @@
+    {"c2pnb272w1", NID_X9_62_c2pnb272w1 },
+    {"c2pnb304w1", NID_X9_62_c2pnb304w1 },
+    {"c2tnb359v1", NID_X9_62_c2tnb359v1 },
+    {"c2pnb368w1", NID_X9_62_c2pnb368w1 },
+    {"c2tnb431r1", NID_X9_62_c2tnb431r1 },
+    /*
+     * the WAP/WTLS curves [unlike SECG, spec has its own OIDs for curves
+     * from X9.62]
+     */
+    {"wap-wsg-idm-ecid-wtls1", NID_wap_wsg_idm_ecid_wtls1 },
+    {"wap-wsg-idm-ecid-wtls3", NID_wap_wsg_idm_ecid_wtls3 },
+    {"wap-wsg-idm-ecid-wtls4", NID_wap_wsg_idm_ecid_wtls4 },
+    {"wap-wsg-idm-ecid-wtls5", NID_wap_wsg_idm_ecid_wtls5 },
+    {"wap-wsg-idm-ecid-wtls6", NID_wap_wsg_idm_ecid_wtls6 },
+    {"wap-wsg-idm-ecid-wtls7", NID_wap_wsg_idm_ecid_wtls7 },
+    {"wap-wsg-idm-ecid-wtls8", NID_wap_wsg_idm_ecid_wtls8 },
+    {"wap-wsg-idm-ecid-wtls9", NID_wap_wsg_idm_ecid_wtls9 },
+    {"wap-wsg-idm-ecid-wtls10", NID_wap_wsg_idm_ecid_wtls10 },
+    {"wap-wsg-idm-ecid-wtls11", NID_wap_wsg_idm_ecid_wtls11 },
+    {"wap-wsg-idm-ecid-wtls12", NID_wap_wsg_idm_ecid_wtls12 },
+    /* IPSec curves */
+    {"Oakley-EC2N-3", NID_ipsec3 },
+    {"Oakley-EC2N-4", NID_ipsec4 },
+    /* brainpool curves */
+    {"brainpoolP160r1", NID_brainpoolP160r1 },
+    {"brainpoolP160t1", NID_brainpoolP160t1 },
+    {"brainpoolP192r1", NID_brainpoolP192r1 },
+    {"brainpoolP192t1", NID_brainpoolP192t1 },
+    {"brainpoolP224r1", NID_brainpoolP224r1 },
+    {"brainpoolP224t1", NID_brainpoolP224t1 },
+    {"brainpoolP256r1", NID_brainpoolP256r1 },
+    {"brainpoolP256t1", NID_brainpoolP256t1 },
+    {"brainpoolP320r1", NID_brainpoolP320r1 },
+    {"brainpoolP320t1", NID_brainpoolP320t1 },
+    {"brainpoolP384r1", NID_brainpoolP384r1 },
+    {"brainpoolP384t1", NID_brainpoolP384t1 },
+    {"brainpoolP512r1", NID_brainpoolP512r1 },
+    {"brainpoolP512t1", NID_brainpoolP512t1 },
+    /* SM2 curve */
+    {"SM2", NID_sm2 },
+};
+
+const char *ec_curve_nid2name(int nid)
+{
+    size_t i;
+
+    if (nid <= 0)
+        return NULL;
+
+    /*
+     * TODO(3.0) Figure out if we should try to find the nid with
+     * EC_curve_nid2nist() first, i.e. make it a priority to return
+     * NIST names if there is one for the NID.  This is related to
+     * the TODO comment in ec_curve_name2nid().
+     */
+
+    for (i = 0; i < OSSL_NELEM(curve_list); i++) {
+        if (curve_list[i].nid == nid)
+            return curve_list[i].name;
+    }
+    return NULL;
+}
+
+int ec_curve_name2nid(const char *name)
+{
+    size_t i;
+    int nid;
+
+    if (name != NULL) {
+        if ((nid = ec_curve_nist2nid_int(name)) != NID_undef)
+            return nid;
+
+#ifndef FIPS_MODULE
+        /*
+         * TODO(3.0) Figure out if we can use other names than the NIST names
+         * ("B-163", "K-163" & "P-192") in the FIPS module, or if other names
+         * are allowed as well as long as they lead to the same curve data.
+         * If only the NIST names are allowed in the FIPS module, we should
+         * move '#endif' to just before 'return NID_undef'.
```

Comment:
There is no reason why non-NIST names for the NIST curves should not be allowed in the FIPS module.  You can drop this TODO.

Reply:
Good, thanks